File: src/main/java/com/eager/questioncloud/workspace/implement/WorkSpaceQuestionRemover.java
package com.eager.questioncloud.workspace.implement;

import com.eager.questioncloud.question.domain.Question;
import com.eager.questioncloud.question.repository.QuestionRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class WorkSpaceQuestionRemover {
    private final QuestionRepository questionRepository;

    public void remove(Long creatorId, Long questionId) {
        Question question = questionRepository.getForModifyAndDelete(questionId, creatorId);
        question.delete();
        questionRepository.save(question);
    }
}

File: qc-core/qc-workspace/qc-workspace-api/src/main/kotlin/com/eager/questioncloud/workspace/implement/WorkspaceQuestionRemover.kt
package com.eager.questioncloud.workspace.implement

import com.eager.questioncloud.question.api.internal.QuestionCommandAPI
import com.eager.questioncloud.workspace.command.DeleteQuestionCommand
import org.springframework.stereotype.Component

@Component
class WorkspaceQuestionRemover(
    private val questionCommandAPI: QuestionCommandAPI,
) {
    fun deleteQuestion(command: DeleteQuestionCommand) {
        questionCommandAPI.delete(command.questionId, command.creatorId)
    }
}